### PR TITLE
Issue 6: Image tags

### PR DESCRIPTION
**Pull request by Peter-Lavigne**
- fixes a potential bug where images are made using `create` and `save` instead of `new` and `save`

- adds tags to the image model
- allows users to add tags when creating an image
- displays tags on the 'show' page for images

### CODE REVIEW

File: test/controllers/images_controller_test.rb [hunk 1]
```diff
@@ -41,4 +41,15 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
       assert_equal 'https://picsum.photos/300/200/?image=1', images[1].attribute('src').value
     end
   end
+
```
Comment (jmangel) [inline test/controllers/images_controller_test.rb]: can we update the `create` test to include a `tag_list`?

**Comment by Peter-Lavigne** [comment]
[(Peter) Issue 6: Image Tags](https://trello.com/c/rgLF2ZXq/43-peter-issue-6-image-tags)

File: test/models/image_test.rb [hunk 2]
```diff
@@ -15,4 +15,12 @@ class ImageTest < ActiveSupport::TestCase
     image = Image.new(url: 'googlecom/image')
     assert_not_predicate(image, :valid?)
   end
+
+  test 'image with multiple tags can be created' do
```
Comment (jmangel) [inline test/models/image_test.rb]: I think there is value in adding a test to make sure that tags are allowed to be blank, assuming that's the desired behavior. At the very least, it adds clarity for developers looking at the tests as to what's allowed

File: .rubocop.yml [hunk 3]
```diff
@@ -6,6 +6,7 @@ AllCops:
     - 'vendor/**/*'
     - 'node_modules/**/*'
     - 'bin/*'
+    - 'db/migrate/*'
```
Comment (jmangel) [inline .rubocop.yml]: This isn't required, but in the way I organize my commits, I would put this in the commit that caused the "issue", which would be the commit where you added all the migration files

File: app/controllers/images_controller.rb [hunk 4]
```diff
@@ -4,7 +4,7 @@ def new
   end
 
   def create
-    @image = Image.create(params.require(:image).permit(:url))
+    @image = Image.new(params.require(:image).permit(:url, :tag_list))
```
Comment (jmangel) [inline app/controllers/images_controller.rb]: Careful, this change includes the addition of `:tag_list` to the parameter, which isn't used until later commits.

Comment (Peter-Lavigne) [reply app/controllers/images_controller.rb]: Whoops, good catch. I'll remember to double check next time

**Pull request merged**